diff --git a/tests/test_itsdangerous/test_timed.py b/tests/test_itsdangerous/test_timed.py
index 18de741..82b4253 100644
--- a/tests/test_itsdangerous/test_timed.py
+++ b/tests/test_itsdangerous/test_timed.py
@@ -54,6 +54,7 @@ class TestTimestampSigner(FreezeMixin, TestSigner):
             signer.unsign(signed)
 
         assert "missing" in str(exc_info.value)
+        assert exc_info.value.date_signed is None
 
     def test_malformed_timestamp(self, signer):
         other = Signer("secret-key")
@@ -63,14 +64,25 @@ class TestTimestampSigner(FreezeMixin, TestSigner):
             signer.unsign(signed)
 
         assert "Malformed" in str(exc_info.value)
+        assert exc_info.value.date_signed is None
 
     def test_future_age(self, signer):
         signed = signer.sign("value")
 
         with freeze_time("1971-05-31"):
-            with pytest.raises(SignatureExpired):
+            with pytest.raises(SignatureExpired) as exc_info:
                 signer.unsign(signed, max_age=10)
 
+        assert isinstance(exc_info.value.date_signed, datetime)
+
+    def test_sig_error_date_signed(self, signer):
+        signed = signer.sign("my string").replace(b"my", b"other", 1)
+
+        with pytest.raises(BadTimeSignature) as exc_info:
+            signer.unsign(signed)
+
+        assert isinstance(exc_info.value.date_signed, datetime)
+
 
 class TestTimedSerializer(FreezeMixin, TestSerializer):
     @pytest.fixture()
